### PR TITLE
Block collector as a worker.

### PR DESCRIPTION
**Pull request by guoger**
Signed-off-by: Jay Guo <guojiannan1101@gmail.com>

### CODE REVIEW

File: pkg/infra/observer.go [hunk 1]
```diff
@@ -78,10 +76,8 @@ func (o *Observer) Start(N int, errorCh chan error, finishCh chan struct{}, now
 		fb := r.Type.(*peer.DeliverResponse_FilteredBlock)
 		o.logger.Debugf("receivedTime %8.2fs\tBlock %6d\tTx %6d\t Address %s\n", time.Since(now).Seconds(), fb.FilteredBlock.Number, len(fb.FilteredBlock.FilteredTransactions), o.Address)
 
-		if blockCollector.Commit(fb.FilteredBlock.Number) {
-			// committed
-			fmt.Printf("Time %8.2fs\tBlock %6d\tTx %6d\t \n", time.Since(now).Seconds(), fb.FilteredBlock.Number, len(fb.FilteredBlock.FilteredTransactions))
-		}
+		// TODO use proper context to create deliver client so it could be cancelled (now it's context.Background).
+		blockCh <- fb.FilteredBlock
```
Comment (SamYuan1990) [inline pkg/infra/observer.go:80]: 这里有个问题，这样改动丢失了block在某个peer上被确认的peer节点信息。

File: pkg/infra/process.go [hunk 2]
```diff
@@ -57,7 +59,8 @@ func Process(configPath string, num int, burst int, rate float64, logger *log.Lo
 
 	start := time.Now()
 
-	go observers.Start(num, errorCh, finishCh, start, blockCollector)
+	go blockCollector.Start(blockCh, finishCh, num, time.Now())
```
Comment (SamYuan1990) [inline pkg/infra/process.go:62]: 我们的goroutine会不会太多了？

Comment (SamYuan1990) [reply pkg/infra/process.go:62]: 比如goroutine之间的切换或者cpu调度会不会影响统计？

**Comment by SamYuan1990** [comment]
before merge, should we remove sync commit?

**Comment by guoger** [comment]
> before merge, should we remove sync commit?

i think it could be kept for reference, at least for a while.

**Review by SamYuan1990 — APPROVED** [review]

**Pull request merged**